src/dialogs/mkdir.tsx

```diff
@@ -67,16 +67,20 @@ const CreateDirectoryModal = ({ currentPath, dialogResult } : {
     const [nameError, setNameError] = useState<string | undefined>();
     const [errorMessage, setErrorMessage] = useState<string | undefined>();
     const [owner, setOwner] = useState<string | undefined>();
+    const [user, setUser] = useState<cockpit.UserInfo| undefined>();
     const createDirectory = () => {
         const path = currentPath + name;
         create_directory(path, owner).then(dialogResult.resolve, err => setErrorMessage(err.message));
     };
-    const id = useId();
     const { cwdInfo } = useFilesContext();
 
+    useEffect(() => {
+        cockpit.user().then(user => setUser(user));
+    }, [setUser]);
```

Comment:
I may misunderstand effect dependencies, but `setUser` is a static function which should *never* change. Also, this is a case for `useInit()`, or if you don't want to use this, an empty dependency array? You can't change the user within a cockpit session.

Reply:
Sure can use `useInit` or an empty depenency array.